deeprank/features/Edesolv.py

```diff
@@ -0,0 +1,411 @@
+        contact = indA + indB
+        
+        # extract the atom keys and xyz of the contact CA atoms
+        keys = self.db.get('serial,chainID,resName,resSeq,name',rowID=contact)
+        xyz = self.db.get('x,y,z',rowID=contact)
+        
+        #Make SASA class
+        sr = SASA.ShrakeRupley()
+        #Compute complex structue SASA
+        sr.compute(struct, level='A') 
+        #Compute free structure SASA
+        sr.compute(free_struct, level='A')
+
+
+        # Get disulfide bonds (important to determine sulfide atoms edesolv values)
+        disulfides_cys = get_disulfide_bonds(self.db)
+
+        
+        # create the dictionary of human readable and xyz-val data
+        self.edesolv_data = {}
+        self.edesolv_data_xyz = {}
+
+        for key, coords in zip(keys, xyz):
+            brk_flag = False
+            disulf_bond = False
+            atom = Atom(key)
+
+            # Calculate complex sasa
+            try:
+                atom.cpx_sasa = struct[0][atom.chainID][atom.resid][atom.name].sasa
+            except KeyError: #Handling alternative residues error
+                print('Alternative residue found at:', key)
+                brk_flag = True
+                pass 
+            
+            # Calculate unbound (translated) sasa
+            try:
+                atom.free_sasa = free_struct[0][atom.chainID][atom.resid][atom.name].sasa
+            except KeyError: #Handling alternative residues error
+                print('Alternative residue found at:', key)
+                brk_flag = True
+                pass 
+            
+            if not brk_flag:
+                
+                # check if the atom is a sulfide involved in a disulfide bond
+                for cys in disulfides_cys:
+                    if cys == (atom.chainID, atom.resid):
+                        disulf_bond = True
+                        
+                # Assign solvation parameters
+                Assign_solv_param(atom, disulf_bond)
+                
+                #Calculate atom complex edesolv, free edesolv and final edesolv (cpx - free)
+                atom.esolcpx = atom.cpx_sasa * atom.solv
+                atom.esolfree = atom.free_sasa * atom.solv
+                atom.edesolv = atom.esolcpx - atom.esolfree
+                esolcpx += atom.esolcpx
+                esolfree += atom.esolfree
+                
+                # xyz-val
+                # { (0|1,x,y,z) : [val] }
+                chain = [{chainA:0,chainB:1}[key[1]]]
+                k = tuple(chain + coords)
+
+                #Human readable
+                self.edesolv_data[(key[1], key[3], key[2], key[4])] = [atom.edesolv]
+                #XYZ data
+                self.edesolv_data_xyz[k] = [atom.edesolv]
+
+        self.feature_data['Edesolv'] = self.edesolv_data
+        self.feature_data_xyz['Edesolv'] = self.edesolv_data_xyz
+
+##################################################################
```

Comment:
You could leave some comments here or remove them

Reply:
Removed in 33d9764